### PR TITLE
Adaptive or responsive design

### PR DESCRIPTION
**Pull request by MaksimkaSota**
Feature:
- Added adaptive or responsive design.

Refactor:
- Some code improvements and optimizations.

### CODE REVIEW

**Comment by ZhekaVasil** [comment]
As I see you use 4 break-points:
max-width: 1440px
max-width: 1280px
max-width: 640px
max-width: 320px

BTW, it is best to use "em" vs "px", please read the article:
https://zellwk.com/blog/media-query-units/


Also you could create sass mixin for the breakpoints, so you dont need to type the numbers every time, there are lot's of examples, just one of them: https://dev.to/thebuildguy/easy-to-write-media-queries-using-sass-mixins-1p2f

File: src/utils/styles/mixins.scss [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+      @content;
+    }
+  }
+  @if $breakpoint==$mobile {
+    @media (max-width: $mobileSize) {
+      @content;
+    }
+  }
+}
+
```
Comment (ZhekaVasil) [inline src/utils/styles/mixins.scss]: I can propose such update:
```
$breakpoints: (
  $mobile: (max-width: $mobileSize),
  $tablet: (max-width: $tabletSize),
  ...
  ...
);


@mixin respond($breakpoint) {
  $query: map-get($breakpoints, $breakpoint);

  @if not $query {
    @error 'No value found for `#{$breakpoint}`. Please make sure it is defined in `$breakpoints` map.';
  }

  @media #{if(type-of($query) == 'string', unquote($query), inspect($query))} {
    @content;
  }
}

```

Comment (ZhekaVasil) [reply src/utils/styles/mixins.scss]: Without if/else statements =)

Comment (MaksimkaSota) [reply src/utils/styles/mixins.scss]: Done!

**Review by ZhekaVasil — APPROVED** [review]

**Pull request merged**